lib/Backend/IR.h

```diff
@@ -503,13 +504,19 @@ class Instr
 
 class ByteCodeUsesInstr : public Instr
 {
-public:
-    static ByteCodeUsesInstr * New(Func * func);
-    static ByteCodeUsesInstr* New(IR::Instr* originalBytecodeInstr, SymID symid);
+private:
     BVSparse<JitArenaAllocator> * byteCodeUpwardExposedUsed;
-    PropertySym *              propertySymUse;
+    
+public:
+    static ByteCodeUsesInstr * New(IR::Instr * originalBytecodeInstr);
+    static ByteCodeUsesInstr * New(Func * originalBytecodeInstr, uint32 offset);
+    const BVSparse<JitArenaAllocator> * getByteCodeUpwardExposedUsed() const { return byteCodeUpwardExposedUsed; }
 
-    void Set(uint symId);
+    PropertySym *              propertySymUse;
+    void Set(bool isjitoptimizedreg, uint symId);
```

Comment:
>bool isjitoptimizedreg, uint symId [](start = 13, length = 34)

nit: you can actually consider getting just the Opnd* as argument - and then you can look up the symId and the isJitOptReg fields.